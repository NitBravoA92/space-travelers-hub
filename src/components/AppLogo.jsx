import logo from '../assets/images/logo.png';

const AppLogo = () => (
  <a href="/" className="logo">
    <img src={logo} alt="Space Travelers&apos; Hub" className="logo-image" />
    <span className="app-name">Space Travelers&apos; Hub</span>
  </a>
);

export default AppLogo;
