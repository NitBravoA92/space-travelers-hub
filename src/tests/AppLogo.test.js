import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AppLogo from '../components/AppLogo';

describe('The AppLogo component', () => {
  test('renders into the DOM correctly', () => {
    const component = renderer
      .create(<AppLogo />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should display the text: Space Travelers\' Hub', () => {
    const component = render(<AppLogo />);

    const expectedText = 'Space Travelers\' Hub';

    expect(component.container).toHaveTextContent(expectedText);
  });

  test('should display an image', () => {
    const component = render(<AppLogo />);

    const logoImage = component.container.querySelector('img');

    expect(logoImage).toHaveAttribute('src', 'logo.png');
    expect(logoImage).toHaveAttribute('alt', 'Space Travelers\' Hub');
  });
});
