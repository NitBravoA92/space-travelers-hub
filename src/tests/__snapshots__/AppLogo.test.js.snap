// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AppLogo component renders into the DOM correctly 1`] = `
<a
  className="logo"
  href="/"
>
  <img
    alt="Space Travelers' Hub"
    className="logo-image"
    src="logo.png"
  />
  <span
    className="app-name"
  >
    Space Travelers' Hub
  </span>
</a>
`;
